### DISCUSSION TITLE
Is brightness supported?

### DISCUSSION THREAD

**Discussion by MrAntonka** [Bodmer/TFT_eSPI#1037] [Q&A - General] [answered]
Is brightness supported? 
ILI9486

Thanks

**Comment by Bodmer** [comment, marked as answer]
Some boards have an onboard transistor to turn on/off the LED, if your board has this then you can control the brightness from your sektch using a PWM signal. The library does not provide this feature as it is so easy to add to a sketch.

**Reply by MrAntonka** [reply]
HI

I know what you are talking about but that is not what I mean.

According to datasheet ILI9486 ( and I believe next drivers has a similar feature too), there is support for brightness:
- Write Display Brightness Value (51h) 
- Write Content Adaptive Brightness Control Value (55h)
- Write CABC Minimum Brightness (5Eh)

I have been looking in the driver init file (ILI9486_Init.h) and I can see there are not any commands to set these registers.
Well, not all displays support this feature, thanks to low prices, manufacturers "optimize" the cost of the display, and the brightness LEDs are powered directly from the LED_POWER pin - that is sad but reality.

But thanks anyway for the reply.
I will let you know when I will know anything new about the brightness. I mean how to drive it via driver.

Roman

**Reply by Bodmer** [reply]
OK, some chips have the internal capability to PWM the external LEDs via a control register and dedicated pins on the control chip. These functions could be accessed by direct register writes but the library does not support this.  I have not seen an assembled display that uses these features, if you have found one (not a bare module) then do post a link here.
